static/docs/user-guide/data-sharing/shared-server.md

```diff
@@ -0,0 +1,107 @@
+# Local Storage on a Shared Development Server
+
+Some teams may prefer using a single shared machine to run their experiments.
+This allows them to have better resource utilization such as the ability to use
+multiple GPUs, etc.
+
+With DVC, you can easily setup a local data storage on the server. This allows
+your team to store and share data of your projects efficiently, and to have
+almost instantaneous data retrieval speed, similar to `git pull` for your code.
+
+## Shared Server Example
+
+Let's see an example of how two different users on the same host can share data
+with the help of a local data storage. So, both of the users and the data
+storage are located on the same machine and no remote server or storage is
+involved.
+
+> For more detailed instructions check out this
+> [interactive example](https://katacoda.com/dvc/courses/examples/shared-server).
+
+<p align="center">
+<img src="/static/img/user-guide/data-sharing/shared-server.png"/>
+</p>
+
+### Setup the server
+
+We need to do these configurations on the server:
+
+- Create accounts for each user and add them to groups for accessing the Git
+  repository and the DVC storage.
+- Create a bare git repository (for example on `/var/local/data/project.git/`)
+  and an empty directory for the DVC storage (for example on
+  `/var/local/data/project.cache/`).
+
+- Grant users read/write access to these directories (through the groups).
+
+### Configure local DVC storage
```

Comment:
this is not how it should be done. The whole purpose of the shared dev server is to share cache not remote. The difference is that cache support different cache link types that saves us from copying files. In that scenario it's pretty common to actually use S3 as a remote storage.

Reply:
> this is not how it should be done. The whole purpose of the shared dev server is to share cache not remote. The difference is that cache support different cache link types that saves us from copying files.

But in a shared development server we can use a deduplicating filesystem (which supports xref copy), so we don't need to resort to cache link types.

Besides this, using a shared cache requires special care when working with commands that modify the cache (for example `dvc gc`). When we use a local DVC storage we don't need to do that, and we just use the normal workflow with `dvc push` and `dvc pull`.

> In that scenario it's pretty common to actually use S3 as a remote storage.

It can still be used to backup and restore the local DVC storage.

Reply:
probably I'm missing something here, I'm not sure. When you do remote DVC won't be using any linking between project's cache and remote right now. Because that's the purpose of having cache - cache is about using reflinks or other link types to avoid duplication and make checkouts fast.

Reply:
> probably I'm missing something here, I'm not sure. When you do remote DVC won't be using any linking between project's cache and remote right now. Because that's the purpose of having cache - cache is about using reflinks or other link types to avoid duplication and make checkouts fast.

If we use a deduplicating filesystem (which supports reflink copy) copying a cached file to the storage (with `dvc push`) will be done instantaneously and it will take 0 (zero) additional space on disk.

Please check again the interactive tutorial: https://katacoda.com/dvc/courses/examples/shared-server It does mention that `dvc push` and `dvc fetch` are very efficient (and it uses a 10G data file as an example).

Reply:
It's a very strong assumption - that all the people share the same mount point. It's in an optimization that we have indeed that works only on certain file systems.

Also, some feedback regarding the tutorial. It's feels super advanced. DVC part starts only in the middle and before that a lot of magic that I think can be simplified.

Reply:
> It's a very strong assumption - that all the people share the same mount point. It's in an optimization that we have indeed that works only on certain file systems.

It's an example that shows how to make things efficient even if the filesystem does not support deduplication (aka reflink copy). It is much more easy to assume that the shared server supports deduplication natively, so no extra configurations or tricks are needed to make things efficient. 

> Also, some feedback regarding the tutorial. It's feels super advanced. DVC part starts only in the middle and before that a lot of magic that I think can be simplified.

Do you think that it would become simpler if the magic that sets up the server and user account is played automatically? This way the people that might be interested in them (linux beginners) would still see them, but the professionals (who already know how to do it) can skip them quickly.

Reply:
> It's an example that shows how to make things efficient even if the filesystem does not support deduplication (aka reflink copy)

I'm not sure how would optimize anything if the filesystem does not support deduplication (aka reflink copy). You would need to share the cache - and it's a completely different setup, right?

>  It is much more easy to assume that the shared server supports deduplication natively, so no extra configurations or tricks are needed to make things efficient.

it's easier, but it's very very specific. Especially the assumption that people work on the same machine + the same FS is way too strong.

> Do you think that it would become simpler if the magic that sets up the server and user account is played automatically? 

I prefer to focus on the actual stuff and hide all auxiliary stuff. There are few mechanisms to hide that - expandable section, extra section at the end with a link to it, FAQ at the end, separate setup section, etc. Not sure what is the best mechanism here. 

> beginners

I'm not sure this is the right audience for these guides in the first place. At least not now. I'm not saying that docs should be complicated, dry and do not provide any help at all. On contrary, even professional would appreciate if we care about them and put some links, materials about setup. It's more about focus, structure, etc.